### ISSUE TITLE
Add `Duration.min()` and `Duration.max()`

### ISSUE THREAD

**Issue by jirutka** [moment/luxon#1527]
Can you please add static methods `min` and `max` to the `Duration`, analogous to the methods in `DateTime`?

**Comment by shashankbhat10** [comment]
Hey @icambron,

If you are ok with adding these methods to the Duration class, can I be assigned this issue?
I have made some changes regarding this and can raise a PR for it.

**Comment by diesieben07** [comment]
This definitely is not trivial. The problem is that durations by default use "casual" conversion rules, which can make it so this comparison would not be reflexive or transitive, depending on which unit you use for comparison.
For example:
29 days > 4 weeks, using days
4 weeks == 1 month, using weeks
29 days < 1 month, using days

So, 29 days is both larger and smaller than 1 month, considering the above logic.

Also: 
4 weeks == 1 month, using weeks; but
4 weeks < 1 month, using days (because 28 days < 30 days)

This gets even more complicated if you have multiple (even different) units on both sides. So really, the user needs to tell which unit to use for the comparison. And even then they might not get the result they want.

**Comment by shashankbhat10** [comment]
I get what you are saying. Would this be somewhat similar to #1517 related to normalization too?

I was assuming, we should be able to compare durations based on their total milliseconds like how DateTimes are compared.
Based on total milliseconds for above scenario, we do get: 4 weeks < 29 days < 1 month.

![duration_compare](https://github.com/moment/luxon/assets/16275922/a86b9256-5837-40c1-9600-8121347e2f88)


I might be missing something while considering all units and negative values in duration as well, but think using milliseconds may be helpful.

**Comment by shashankbhat10** [comment]
I get what you are saying. I think could be done after any resolution of #1514, right?

**Comment by diesieben07** [comment]
No, this is an inherent property of the casual conversion matrix that Durations use by default. You can read more about it in [the documentation](https://moment.github.io/luxon/#/math?id=casual-vs-longterm-conversion-accuracy).
